### PR TITLE
Mainnet full node opt in

### PR DESCRIPTION
**Pull request by tanx**
This change should allow users to opt into mainnet when running a full node, sidestepping the neutrino dependency in the short term. Obviously this required about 200 GB of storage. But it will allow us to start testing the app more thoroughly before our neutrino launch.

### CODE REVIEW

**Comment by tanx** [comment]
I've finally gotten my local bitcoind node synced to mainnet and have been able to test this. After seed backup, the app currently gets stuck on the wait spinner. Running the tagged lnd v0.5.2-beta release.

I start bitcoind like so (with empty bitcoin.conf):

```
bitcoind -txindex=1 -rpcuser=kek -rpcpassword=kek -rpcbind=localhost -zmqpubrawblock=tcp://127.0.0.1:28332 -zmqpubrawtx=tcp://127.0.0.1:28333
```

I start the app like so:

```
./dist/mac/Lightning.app/Contents/MacOS/Lightning --bitcoin.mainnet --bitcoin.node=bitcoind --bitcoind.rpcuser=kek --bitcoind.rpcpass=kek --bitcoind.zmqpubrawblock=tcp://127.0.0.1:28332 --bitcoind.zmqpubrawtx=tcp://127.0.0.1:28333
```

Here are the logs:

```


 ___       ________       ________  ________  ________
|\  \     |\   ___  \    |\   __  \|\   __  \|\   __  \
\ \  \    \ \  \\ \  \   \ \  \|\  \ \  \|\  \ \  \|\  \
 \ \  \    \ \  \\ \  \   \ \   __  \ \   ____\ \   ____\
  \ \  \____\ \  \\ \  \   \ \  \ \  \ \  \___|\ \  \___|
   \ \_______\ \__\\ \__\   \ \__\ \__\ \__\    \ \__\
    \|_______|\|__| \|__|    \|__|\|__|\|__|     \|__|



(electron) 'app.makeSingleInstance(cb)' is deprecated. Use 'app.requestSingleInstanceLock() and app.on('second-instance', cb)' instead.
[12:35:50.893] [info] Using lnd in path lnd
[12:35:51.156] [info] lnd: 2019-02-12 12:35:51.155 [WRN] LTND: open /Users/tanx/Library/Application Support/lightning-app/lnd/lnd.conf: no such file or directory

[12:35:51.157] [info] lnd: 2019-02-12 12:35:51.155 [INF] LTND: Version: 0.5.2-beta commit=v0.5.2-beta, build=production, logging=default
2019-02-12 12:35:51.155 [INF] LTND: Active chain: Bitcoin (network=mainnet)

[12:35:51.159] [info] lnd: 2019-02-12 12:35:51.159 [INF] CHDB: Checking for schema update: latest_version=7, db_version=7

[12:35:51.159] [info] lnd: 2019-02-12 12:35:51.159 [INF] RPCS: Generating TLS certificates...

[12:35:51.174] [info] lnd: 2019-02-12 12:35:51.174 [INF] RPCS: Done generating TLS certificates

[12:35:51.179] [info] lnd: 2019-02-12 12:35:51.178 [INF] RPCS: password RPC server listening on 127.0.0.1:10006

[12:35:51.181] [info] lnd: 2019-02-12 12:35:51.180 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8086
2019-02-12 12:35:51.181 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.

[12:35:51.347] [info] Loaded initial state
[12:35:51.905] [info] GRPC unlockerReady
[12:35:52.134] [info] Saved state
[12:35:52.618] [info] Saved state
[12:36:45.814] [info] lnd: 2019-02-12 12:36:45.814 [INF] LNWL: Opened wallet

[12:36:45.871] [info] lnd: 2019-02-12 12:36:45.871 [INF] LTND: Primary chain is set to: bitcoin

[12:36:45.892] [info] lnd: 2019-02-12 12:36:45.891 [INF] LTND: Initializing bitcoind backed fee estimator
2019-02-12 12:36:45.891 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on tcp://127.0.0.1:28332

[12:36:45.892] [info] lnd: 2019-02-12 12:36:45.891 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on tcp://127.0.0.1:28333

[12:36:46.604] [info] lnd: 2019-02-12 12:36:46.604 [INF] LNWL: The wallet has been unlocked without a time limit

[12:36:46.614] [info] lnd: 2019-02-12 12:36:46.613 [INF] LTND: LightningWallet opened

[12:36:46.622] [info] lnd: 2019-02-12 12:36:46.622 [INF] LNWL: Caught up to height 0

[12:36:46.624] [info] lnd: 2019-02-12 12:36:46.624 [INF] HSWC: Restoring in-memory circuit state from disk

[12:36:46.624] [info] lnd: 2019-02-12 12:36:46.624 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0

[12:36:46.630] [info] lnd: 2019-02-12 12:36:46.630 [INF] RPCS: RPC server listening on 127.0.0.1:10006

[12:36:46.632] [info] lnd: 2019-02-12 12:36:46.632 [INF] RPCS: gRPC proxy started at 127.0.0.1:8086

[12:36:46.634] [info] lnd: 2019-02-12 12:36:46.634 [INF] LTND: Waiting for chain backend to finish sync, start_height=562684

[12:36:48.567] [info] GRPC unlockerClosed
[12:36:48.571] [info] GRPC lndReady
[12:36:48.575] [error] Caught Main Process Error:
[12:36:48.584] [error] Caught Main Process Error:
[12:36:48.585] [error] Caught Main Process Error:
[12:36:48.585] [error] Caught Main Process Error:
[12:36:48.586] [error] Caught Main Process Error:
[12:36:48.586] [error] Caught Main Process Error:
[12:36:48.586] [error] Caught Main Process Error:
[12:36:48.587] [error] Caught Main Process Error:
[12:36:48.587] [error] Caught Main Process Error:
[12:36:48.588] [error] Caught Main Process Error:
[12:36:48.588] [error] Caught Main Process Error:
[12:36:48.591] [error] Caught Main Process Error:
[12:36:48.591] [error] Caught Main Process Error:
[12:36:48.953] [info] Saved state
[12:37:29.515] [info] lnd: 2019-02-12 12:37:29.515 [INF] LNWL: Caught up to height 10000

[12:38:12.572] [info] lnd: 2019-02-12 12:38:12.572 [INF] LNWL: Caught up to height 20000

[12:38:55.321] [info] lnd: 2019-02-12 12:38:55.320 [INF] LNWL: Caught up to height 30000

[12:40:06.983] [info] lnd: 2019-02-12 12:40:06.971 [INF] LNWL: Caught up to height 40000

[12:40:48.814] [info] lnd: 2019-02-12 12:40:48.814 [INF] LNWL: Caught up to height 50000

[12:41:30.167] [info] lnd: 2019-02-12 12:41:30.166 [INF] LNWL: Caught up to height 60000

[12:41:48.589] [error] Getting new wallet address failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.594] [error] Getting wallet balance failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.595] [error] Getting channel balance failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.595] [error] Listing peers failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.596] [error] Listing channels failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.597] [error] Listing pending channels failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.597] [error] Listing closed channels failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.598] [error] Listing transactions failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.598] [error] Listing invoices failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.599] [error] Listing payments failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:48.600] [error] Getting node info failed { code: 4,
  details: 'Deadline Exceeded',
  metadata: { _internal_repr: {} } }
[12:41:49.990] [error] Caught Main Process Error:
[12:41:49.991] [error] Caught Main Process Error:
[12:41:49.992] [error] Caught Main Process Error:
[12:42:38.225] [info] lnd: 2019-02-12 12:42:38.224 [INF] LNWL: Caught up to height 70000

^C[12:42:39.188] [info] lnd: 2019-02-12 12:42:39.162 [INF] LTND: Received SIGINT (Ctrl+C).
2019-02-12 12:42:39.163 [INF] LTND: Shutting down...
2019-02-12 12:42:39.163 [INF] LTND: Gracefully shutting down.
```

@Roasbeef any idea what I'm doing wrong?

File: README.md [hunk 1]
```diff
@@ -77,25 +77,25 @@ To run the packaged version of the app e.g. for macOS run:
...
-To run the packaged version of the app e.g. for macOS run:
+To run the packaged version of the app e.g. for macOS run (set `--bitcoin.mainnet` for mainnet):
 ```
-./dist/mac/Lightning.app/Contents/MacOS/Lightning --btcd.rpcuser=kek --btcd.rpcpass=kek
+./dist/mac/Lightning.app/Contents/MacOS/Lightning --bitcoin.testnet --btcd.rpcuser=kek --btcd.rpcpass=kek
 ```
 
 #### bitcoind
-Start bitcoind in a seperate terminal session and wait until it's fully synced (can take over a day)
+Start bitcoind in a seperate terminal session and wait until it's fully synced (can take over a day). Remove the `-testnet` flag for mainnet:
```
[the remark this answers is not in the document]
Comment (tanx) [reply README.md]: Rebased on the current master and fixed the typos.

**Comment by tanx** [comment]
> Can it really be this easy to enable mainnet on desktop?? 😈

Right?! 😃

> Tbh I'm still waiting for my `btcd` and `bitcoind` nodes to sync mainnet. I'd prefer to test to make sure that switching back and forth between testnet and mainnet works seamlessly.
> 
> But, I'm fine if you want to merge this with a `utACK`, since it looks good to me.

I'm currently syncing my btcd node on mainnet as well. We should definitely test this before merging the PR.

**Comment by Roasbeef** [comment]
@tanx I think it's clarified, but the sync for bitcoind is a bit slow since it walks up the entire chain and stores each header. This is useful since it lets us switch between full nodes backends easily, but in the future, it'll be near instant as we'll just wait for the node to sync, then store the current chain tip.

**Comment by Roasbeef** [comment]
While it's syncing to the node, the main RPC server won't be on yet, but `getinfo` _should_ work.

**Comment by tanx** [comment]
> While it's syncing to the node, the main RPC server won't be on yet, but `getinfo` _should_ work.

Ah ok! That's probably why all the grpc calls fail. The app stays on the wait spinner until there is a walletAddress set in the mobx store.

**Comment by tanx** [comment]
@valentinewallace I tested/fixed this and the PR should be good to merge once we have the new btcd clusters in place.

**Pull request merged**